demos/src/aws_iot_demo_provisioning.c

```diff
@@ -415,10 +415,10 @@ int RunProvisioningDemo( bool awsIotMqttMode,
 
     /* The list of parameters that will be used as "context" for provisioning the demo application.
      * This demo will pass exactly 2 different parameter entries for requesting provisioning. */
-    AwsIotProvisioningRequestParameterEntry_t provisioningParameters[ 2 ] = { 0 };
+    AwsIotProvisioningRequestParameterEntry_t provisioningParameters[ 2 ] = { { 0 } };
 
     /* Determine if a provisioning template name has been specified. */
-    if( ( pTemplateName == NULL ) || ( strlen( pTemplateName ) == 0 ) )
+    if( ( &pTemplateName[ 0 ] == NULL ) || ( strlen( pTemplateName ) == 0 ) )
```

Comment:
& and [0] seems kind of like a hack, especially for a static variable. Since the default initialization is to "", is the first condition even necessary?

Reply:
I agree that it is a hack. Both Dan Griffin and Cobus wanted the NULL check to exist as the value of the #define is customer configurable. (In a negative use case, one can remove the default `""` with nothing)

Reply:
Adding to it, if the pointer value were NULL, wouldn't [ 0 ] cause an error as it dereferences the pointer? I don't think this part needs to be here.

Reply:
Would casting the array to ( void * ) before the comparison work?